### PR TITLE
Unit test cases for close position, add collateral and claim rewards

### PR DESCRIPTION
**Pull request by avkr003**

### CODE REVIEW

**Comment by avkr003** [comment]
The software upgrade workflow is unable to find the genesis file:
panic: couldn't read GenesisDoc file: open /home/runner/.elys2/config/genesis.json: no such file or directory

File: x/leveragelp/keeper/msg_server_add_collateral_test.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	return
+}
+
+func (suite *KeeperTestSuite) TestMsgServerAddCollateral() {
+	addresses := simapp.AddTestAddrs(suite.app, suite.ctx, 10, sdk.NewInt(1000000))
+	asset1 := ptypes.ATOM
+	asset2 := ptypes.BaseCurrency
+	leverage := sdk.MustNewDecFromStr("2.0")
+	collateralAmount := sdk.NewInt(10000000)
+	//leverageLPShares := sdk.MustNewDecFromStr("20000095238095238100").TruncateInt()
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_add_collateral_test.go:105]:
```suggestion
```

File: x/leveragelp/keeper/msg_server_add_collateral_test.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+				Collateral: sdk.NewInt(10000),
+			},
+			false,
+			"",
+			func() {
+				suite.ResetSuite()
+				SetupCoinPrices(suite.ctx, suite.app.OracleKeeper)
+				initializeForAddCollateral(suite, addresses, asset1, asset2, true)
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+			},
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_add_collateral_test.go:210]: is it possible to verify here if collateral amount has increased and lpTotal amount remains same after adding collateral ?

File: x/leveragelp/keeper/msg_server_claim_rewards_test.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				openPosition(suite, addresses[0], collateralAmount, leverage)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			tc.prerequisiteFunction()
+			msgServer := keeper.NewMsgServerImpl(suite.app.LeveragelpKeeper)
+			_, err := msgServer.ClaimRewards(suite.ctx, tc.input)
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_claim_rewards_test.go:158]: same, can we verify claim rewards numbers ?

File: x/leveragelp/keeper/msg_server_claim_rewards_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	}
+}
+
+func (suite *KeeperTestSuite) TestMsgServerClaimRewards() {
+	addresses := simapp.AddTestAddrs(suite.app, suite.ctx, 10, sdk.NewInt(1000000))
+	asset1 := ptypes.ATOM
+	asset2 := ptypes.BaseCurrency
+	leverage := sdk.MustNewDecFromStr("2.0")
+	collateralAmount := sdk.NewInt(10000000)
+	//leverageLPShares := sdk.MustNewDecFromStr("20000095238095238100").TruncateInt()
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_claim_rewards_test.go:90]:
```suggestion
```

File: x/leveragelp/keeper/msg_server_close_test.go [hunk 5]
```diff
@@ -171,6 +259,17 @@ func (suite *KeeperTestSuite) TestClose() {
...
+			&types.MsgClose{
+				Creator:  addresses[0].String(),
+				Id:       1,
+				LpAmount: sdk.NewInt(0),
+			},
+			false,
+			"",
+			func() {
+			},
+		},
```
Comment (amityadav0) [inline x/leveragelp/keeper/msg_server_close_test.go:272]: same, when we close position verify that lpAmount closed is correct.

File: x/leveragelp/keeper/msg_server_add_collateral_test.go (hunk 2 above)
Comment (avkr003) [reply x/leveragelp/keeper/msg_server_add_collateral_test.go:209]: Yes, let me change the unit test structure and add post validations

**Review by cosmic-vagabond — APPROVED** [review]

**Pull request merged**